### PR TITLE
Add UDP demo running FreeRTOS+TCP

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR adds a UDP demo as mentioned [here ](https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_UDP/Embedded_Ethernet_Examples/RTOS_UDP_CLI_Windows_Simulator.html). The UDP demo use FreeRTOS+TCP stack in UDP-only mode.

Furthermore, the folder of the old UDP demo is removed which used FreeRTOS+UDP which got replaced by FreeRTOS+TCP after FreeRTOS V10.1.0. The demo folder is present [here](https://github.com/FreeRTOS/FreeRTOS/tree/152b7db2c1f146a181d75798d10e3840281c9fa5/FreeRTOS-Plus/Demo/FreeRTOS_Plus_UDP_and_CLI_Windows_Simulator) as of [this](https://github.com/FreeRTOS/FreeRTOS/commit/152b7db2c1f146a181d75798d10e3840281c9fa5) commit.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by alfred2g — DISMISSED** [review]

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_UDP_Mode_CLI_Windows_Simulator/WinPCap/arch.c [hunk 1]
```diff
@@ -0,0 +1,377 @@
+/*
+    FreeRTOS V7.3.0 - Copyright (C) 2012 Real Time Engineers Ltd.
+
```
Comment (cobusve) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_UDP_Mode_CLI_Windows_Simulator/WinPCap/arch.c:3]: This is for sure not the right header!

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_UDP_Mode_CLI_Windows_Simulator/WinPCap/netif.h [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
+    FreeRTOS V7.3.0 - Copyright (C) 2012 Real Time Engineers Ltd.
```
Comment (cobusve) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_UDP_Mode_CLI_Windows_Simulator/WinPCap/netif.h:2]: This header also needs updating

**Review by cobusve — CHANGES_REQUESTED** [review]
There are some outdated headers in this one I marked 2 of them, the header check is also failing

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_UDP_Mode_CLI_Windows_Simulator/WinPCap/arch.c (hunk 1 above)
Comment (AniruddhaKanhere) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_UDP_Mode_CLI_Windows_Simulator/WinPCap/arch.c]: I removed the WinPCap folder altogether.

**Review by cobusve — APPROVED** [review]

**Pull request merged**